### PR TITLE
fix!: changed rollup to tsup to simplify bundling

### PR DESCRIPTION
**Pull request by gentlementlegen**
Relates to https://github.com/ubiquity-os-marketplace/text-conversation-rewards/pull/144

### CODE REVIEW

**Review by 0x4007 — COMMENTED** [review]
Pretty sure esbuild is superior what are you trying to accomplish here

**Comment by gentlementlegen** [comment]
@0x4007 `tsup` is literally built on top of `esbuild`, we use it on most of out projects and it helps with bundling configuration made much simpler.

**Comment by gentlementlegen** [comment]
https://github.com/egoist/tsup?tab=readme-ov-file#tsup

Needed to work within ESM projects as files were not compiled properly.

**Review by 0x4007 — APPROVED** [review]

File: tsup.config.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { defineConfig } from "tsup";
+
+export default defineConfig({
+  entry: ["src/index.ts", "src/core/index.ts", "src/utils/index.ts", "src/handlers/index.ts", "src/types/index.ts"],
+  splitting: false,
```
Comment (0x4007) [inline tsup.config.ts:5]: Tree shaking seems useful is that what this is

File: tsup.config.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  sourcemap: false,
```
Comment (0x4007) [inline tsup.config.ts:6]: This also seems very useful
```suggestion
  sourcemap: true,
```

**Comment by 0x4007** [comment]
You can merge after you make a decision on my review comment

**Comment by 0x4007** [comment]
You can merge after you make a decision on my review comment

File: tsup.config.ts (hunk 2 above)
Comment (gentlementlegen) [reply tsup.config.ts]: Not source that sourcemap are useful except for stack traces and debug which should not be included in production. It is useful if the code is minimized, which is not the case here.

File: tsup.config.ts (hunk 1 above)
Comment (gentlementlegen) [reply tsup.config.ts:5]: You mean `splitting` or `tree-shaking`?
Splitting is an option only relevant for `ESM`, not `CJS` and is not handled by `esbuild` which is why it's disabled. `treeShaking` is default on.
https://tsup.egoist.dev/#code-splitting
https://tsup.egoist.dev/#tree-shaking

File: tsup.config.ts (hunk 2 above)
Comment (gentlementlegen) [reply tsup.config.ts]: See also https://tsup.egoist.dev/#generate-sourcemap-file

Comment (0x4007) [reply tsup.config.ts]: > Not sure that sourcemap are useful except for stack traces and debug which should not be included in production. It is useful if the code is minimized, which is not the case here.

Stack traces aren't magically removed from production I see them all the time from packages

**Pull request merged**